packages/commerce-sdk-react/src/auth/storage.ts

```diff
@@ -44,16 +45,28 @@ export class CookieStorage extends BaseStorage {
         super(options)
 
         if (typeof document === 'undefined') {
-            throw new Error('CookieStorage is not avaliable on the current environment.')
+            throw new Error('CookieStorage is not available on the current environment.')
         }
     }
     set(key: string, value: string, options?: Cookies.CookieAttributes) {
         const suffixedKey = this.getSuffixedKey(key)
+        const parentUrl = document.location?.ancestorOrigins?.[0] || document.referrer
+        const parentHostName = parentUrl ? new URL(parentUrl).hostname : ''
+        const isInAllowList = IFRAME_HOST_ALLOW_LIST.includes(parentHostName)
```

Comment:
```suggestion
        const isParentIframeTrusted = IFRAME_HOST_ALLOW_LIST.includes(parentHostName)
```
Not super important, but I read `!isLocalHost && isInAllowList` and thought "wait, do we need two checks? localhost isn't in the allowlist?". Took me a minute to remember that `isLocalHost` checks the storefront and `isInAllowList` checks the parent. A different name might help with that. 🤷